### PR TITLE
feat(ios/app): universal links for keyboard installation

### PR DESCRIPTION
**Pull request by jahorton**
Continuing on from #3465, this fine fellow gets universal links connected and working.

In my current local tests, I have to longpress the links (to get an "Open in Keyman" option), but this is due to my local machine's refusal to auto-update to the latest provisioning profile.  Build environment matters here.

### CODE REVIEW

File: ios/engine/KMEI/KeymanEngineTests/UniversalLinkTests.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    parsedLink = UniversalLinks.tryParseKeyboardInstallLink(URL.init(string: "\(KeymanHosts.KEYMAN_COM)/keyboards/install/sil_euro_latin")!)
+
+    if let parsedLink = parsedLink {
+      XCTAssertEqual(parsedLink.keyboard_id, "sil_euro_latin")
+      XCTAssertNil(parsedLink.lang_id)
+    }
+
+    parsedLink = UniversalLinks.tryParseKeyboardInstallLink(URL.init(string: "\(KeymanHosts.KEYMAN_COM)/keyboards/install/foo?bcp47=bar")!)
```
Comment (mcdurdin) [inline ios/engine/KMEI/KeymanEngineTests/UniversalLinkTests.swift:39]: Can we add tests for urls such as

```
\(KeymanHosts.KEYMAN_COM)/keyboards/install/foo?_t=123&bcp47=bar
\(KeymanHosts.KEYMAN_COM)/keyboards/install/foo?bcp47=bar&baz=nope
```

File: ios/keyman/Keyman/Keyman/Keyman.entitlements [hunk 2]
```diff
@@ -2,6 +2,11 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>com.apple.developer.associated-domains</key>
+	<array>
+		<string>applinks:keyman.com</string>
+		<string>applinks:keyman-staging.com</string>
```
Comment (mcdurdin) [inline ios/keyman/Keyman/Keyman/Keyman.entitlements:8]: Do we need to have something that restricts to a specific path here?

Comment (jahorton) [reply ios/keyman/Keyman/Keyman/Keyman.entitlements:8]: Nope.  That's what https://github.com/keymanapp/keyman.com/pull/141 was for.  I think the in-app entitlements stuff says "check these sites for the apple-app-site-association file", or something like that.

Well, that and https://github.com/keymanapp/keyman/pull/3466/files#diff-42c700751fafa933a17969ebb675653fR102-R139.

File: ios/engine/KMEI/KeymanEngineTests/UniversalLinkTests.swift (hunk 1 above)
Comment (jahorton) [reply ios/engine/KMEI/KeymanEngineTests/UniversalLinkTests.swift:39]: I assume these tests should 'pass' - that `("foo", "bar")` should result from both, with the other parameters ignored?

Comment (mcdurdin) [reply ios/engine/KMEI/KeymanEngineTests/UniversalLinkTests.swift:39]: Correct

**Pull request merged**